WordPress/wordpress-playground
# Fix manifest.json URLs

## Motivation for the change, related issues

The manifest.json file is currently broken which prevents users from installing Playground as a PWA.

## Implementation details

The manifest.json file requires full paths, so we need different versions for each environment.

When building for production, we copy the manifest.json file in the dist folder. This is the default build behavior.

If the build runs in the development mode, it will modify the manifest.json file url
to the `developmentBuildUrl` and copy it to the dist folder.
To trigger a development build, run `npx nx run playground-website:build:wasm-wordpress-net:development`.

The development server needs it's own version of the manifest, so we modify the URL to use the `localServerUrl`.

## Testing Instructions (or ideally a Blueprint)

- checkout this branch
- start dev server
- [Confirm that manifest.json](http://127.0.0.1:5400/website-server/manifest.json) points to _http://127.0.0.1:5400/website-server/_
- Run `npx nx run playground-website:build:wasm-wordpress-net`
- Confirm that `dist/packages/playground/wasm-wordpress-net/manifest.json` points to _https://playground.wordpress.net/_
- Run `PLAYGROUND_URL='http://localhost:9999/' npx nx run playground-website:build:wasm-wordpress-net`
- Confirm that `dist/packages/playground/wasm-wordpress-net/manifest.json` points to _https://localhost:9999/_
